src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py

```diff
@@ -515,7 +509,12 @@ def getJobParameters(self, jobList):
             ad['Arguments'] = "%s %i %s" % (os.path.basename(job['sandbox']), job['id'], job["retry_count"])
             ad['transfer_output_files'] = "Report.%i.pkl,wmagentJob.log" % job["retry_count"]
 
-            # Do not define Requirements and X509 ads for Volunteer resources
+            # Dictionary keys need to be consistent across all jobs within the same 
+            # clusterId when working with queue_with_itemdata()
+            # Initialize 'Requirements' to an empty string for all jobs.
+            # See issue: https://htcondor-wiki.cs.wisc.edu/index.cgi/tktview?tn=7715 
+            ad['Requirements'] = ''
+            # Do not define custom Requirements for Volunteer resources
             if self.reqStr and "T3_CH_Volunteer" not in job.get('possibleSites'):
```

Comment:
Should we simply set it if `self.reqStr is not None`?

Reply:
`ad['Requirements']` needs to be set always. This is what I recently learned from the HTCondor ticket. The itemdata object in condor creates a "table", so if we set a classad in a job, all the other jobs within the same clusterId need to have it set to something too, even if it's just an empty string, as in this case.

Reply:
I see. What I wanted to say was something like:
```
ad['Requirements'] = ''
if self.reqStr is not None:
                ad['Requirements'] = self.reqStr
```

Which means, if the agent is instructed to define a requirement string, just define it regardless of the sites in the whitelist. It's cleaner and provides a better performance, but I do agree that, one day in future, it might cause us problems. Just an idea, I don't really have any preference here.

Reply:
Gotcha! Yes, I agree it is cleaner. Whoever defines `reqStr` in the future can exclude a specific site in the expression anyway (e.g.: `(GLIDEIN_CMSSite =!= "T3_CH_Volunteer") && (more conditions)`), so I will follow your suggestion.